extensions/panache/mongodb-panache-kotlin/deployment/src/main/java/io/quarkus/mongodb/panache/kotlin/deployment/visitors/KotlinPanacheMongoRepositoryClassVisitor.java

```diff
@@ -0,0 +1,107 @@
+package io.quarkus.mongodb.panache.kotlin.deployment.visitors;
+
+import static io.quarkus.deployment.util.AsmUtil.getDescriptor;
+import static io.quarkus.mongodb.panache.kotlin.deployment.KotlinGenerator.findEntityTypeArguments;
+
+import java.lang.reflect.Modifier;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.jboss.jandex.AnnotationValue;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.IndexView;
+import org.jboss.jandex.MethodInfo;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.MethodVisitor;
+
+import io.quarkus.mongodb.panache.deployment.ByteCodeType;
+import io.quarkus.mongodb.panache.deployment.TypeBundle;
+import io.quarkus.mongodb.panache.kotlin.deployment.KotlinGenerator;
+import io.quarkus.panache.common.deployment.PanacheEntityEnhancer;
+import io.quarkus.panache.common.deployment.visitors.PanacheRepositoryClassVisitor;
+
+public class KotlinPanacheMongoRepositoryClassVisitor extends PanacheRepositoryClassVisitor {
+    private final TypeBundle types;
+    private KotlinGenerator generator;
+    final Map<String, MethodInfo> toGenerate = new TreeMap<>();
+    final Map<String, MethodInfo> toElide = new TreeMap<>();
+
+    public KotlinPanacheMongoRepositoryClassVisitor(IndexView indexView, ClassVisitor outputClassVisitor, String className,
+            TypeBundle types) {
+        super(className, outputClassVisitor, indexView);
+        this.types = types;
+    }
+
+    @Override
+    protected final DotName getPanacheRepositoryDotName() {
+        return types.repository().dotName();
+    }
+
+    @Override
+    protected final DotName getPanacheRepositoryBaseDotName() {
+        return types.repositoryBase().dotName();
+    }
+
+    @Override
+    protected final String getPanacheOperationsInternalName() {
+        return types.operations().internalName();
+    }
+
+    @Override
+    public void visit(int version, int access, String name, String signature, String superName, String[] interfaces) {
+        super.visit(version, access, name, signature, superName, interfaces);
+        final String repositoryClassName = name.replace('/', '.');
+
+        ByteCodeType[] foundTypeArguments = findEntityTypeArguments(indexView, repositoryClassName,
+                getPanacheRepositoryBaseDotName());
+
+        ByteCodeType idType = foundTypeArguments[1].unbox();
```

Comment:
In this case, yes, because the `Id` parameter has an upper bound of `Any` and not `Any?` which means the actual type used in method signatures would be the primitive types and not the wrappers.  So it would be safe to `unbox()` there because any non-wrapper type just gets passed through while the wrappers would get unboxed.